### DISCUSSION TITLE
public文件夹导致eslint时内存泄漏

### DISCUSSION THREAD

**Discussion by songshuzhong** [umijs/umi#12539] [Q&A]
如题，当public文件夹下静态资源很多时，虽然在ignore里忽略了public目录但还是影响到了eslint，导致内存泄漏。请问是什么原因？

**Comment by fz6m** [comment]
你的意思是 public 一大堆 js 文件，但是 eslint 却依然会扫描 public 下的 js 文件格式吗，用 `.eslintignore` 忽略一下吧。

**Reply by songshuzhong** [reply]
ignore文件配置：
------------------------------------------------------------------------------------------
{
  "ignorePatterns": [
    "**/dist/**/*",
    "**/lib/**/*",
    "**/node_modules/**/*",
    "apps/playground/public/"
  ]
}
报错信息：
------------------------------------------------------------------------------------------
[STARTED] Preparing lint-staged...
[COMPLETED] Preparing lint-staged...
[STARTED] Running tasks for staged files...
[STARTED] package.json — 8 files
[STARTED] *.{js,jsx,ts,tsx} — 8 files
[STARTED] eslint --fix
[FAILED] eslint --fix [FAILED]
[FAILED] eslint --fix [FAILED]
[COMPLETED] Running tasks for staged files...
[STARTED] Applying modifications from tasks...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
[COMPLETED] Reverting to original state because of errors...
[STARTED] Cleaning up temporary files...
[COMPLETED] Cleaning up temporary files...

× eslint --fix:

<--- Last few GCs --->

[9420:000002AE3A5DDEF0]    30102 ms: Mark-sweep (reduce) 2046.2 (2084.6) -> 2045.1 (2082.9) MB, 291.3 / 0.0 ms  (+ 83.8 ms in 18 steps since start of marking, biggest step 10.8 ms, walltime since start of marking 388 ms) (average mu = 0.520, current mu = [9420:000002AE3A5DDEF0]    30632 ms: Mark-sweep (reduce) 2046.3 (2082.9) -> 2045.7 (2083.6) MB, 527.8 / 0.0 ms  (average mu = 0.309, current mu = 0.005) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF7AA775F6F node_api_throw_syntax_error+174175
 2: 00007FF7AA6F9356 v8::internal::wasm::WasmCode::safepoint_table_offset+59926
 3: 00007FF7AA6FAFF0 v8::internal::wasm::WasmCode::safepoint_table_offset+67248
 4: 00007FF7AB1A6754 v8::Isolate::ReportExternalAllocationLimitReached+116
 5: 00007FF7AB191AE2 v8::Isolate::Exit+674
 6: 00007FF7AB0137AC v8::internal::EmbedderStackStateScope::ExplicitScopeForTesting+124
 7: 00007FF7AB0109CB v8::internal::Heap::CollectGarbage+3963
 8: 00007FF7AB026C03 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath+2099
 9: 00007FF7AB0274AD v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath+93
10: 00007FF7AB036C70 v8::internal::Factory::NewFillerObject+816
11: 00007FF7AAD27615 v8::internal::DateCache::Weekday+1349
12: 00007FF7AB244651 v8::internal::SetupIsolateDelegate::SetupHeap+558257
13: 00007FF72B669C01
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
husky - pre-commit hook exited with code 1 (error)

**Reply by fz6m** [reply]
你这是在 monorepo 根目录配置的 eslint 吗，eslint 需要配置在每个项目的目录，然后自己研究下 eslint 的 ignore 配置吧，用 glob 模式的，可能你写的这种没生效，参考官方文档：https://eslint.org/docs/v8.x/use/configure/ignore#ignorepatterns-in-config-files
